src/app/shared/filter-bar/label-filter-bar/label-filter-bar.component.ts

```diff
@@ -93,4 +93,10 @@ export class LabelFilterBarComponent implements OnInit, OnDestroy {
   filter(filter: string, target: string): boolean {
     return !target.toLowerCase().includes(filter.toLowerCase());
   }
+
+  removeAllSelection(): void {
+    this.selectedLabels.next([]);
+    this.selectedLabelNames = [];
+    this.cdref.detectChanges();
+  }
```

Comment:
```suggestion
  @ViewChild(MatSelectionList) matSelectionList;   //add after L19
  
  removeAllSelection(): void {
    this.matSelectionList.deselectAll();
  }
```

Reply:
after spending a few hours investigating this, I thought it might be interesting to share what caused this `ExpressionChanged...Error`.

Angular throws an `ExpressionChangedAfterItHasBeenCheckedError` when an expression value has been changed after change detection has completed. (see: https://angular.io/errors/NG0100)

To begin you must first understand the stages in the component lifecycle (which includes change detection) - https://angular.io/guide/lifecycle-hooks#prerequisites

In this case, this is probably caused by a child component changing a parent's property. The parent would be `LabelFilterBarComponent`, the child would be `MatSelectionList`. The change detection is initiated by `this.selectedLabelNames = []`. The child then changes this property again through the two-way binding of `ngModel`:

```
<mat-selection-list [(ngModel)]="selectedLabelNames (selectionChange)="selectedLabels.next(selectedLabelNames)">
```

Reply:
I think it's better not to use cdref, and in this case we don't need it